app/kuma-dp/pkg/dataplane/accesslogs/server.go

```diff
@@ -0,0 +1,121 @@
+package accesslogs
+
+import (
+	"fmt"
+	"github.com/Kong/kuma/pkg/config/app/kuma-dp"
+	"github.com/Kong/kuma/pkg/core"
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/envoyproxy/go-control-plane/envoy/data/accesslog/v2"
+	v2 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var logger = core.Log.WithName("accesslogs")
+
+type accessLogServer struct {
+	server *grpc.Server
+	conn   net.Conn
+}
+
+func NewAccessLogServer() *accessLogServer {
+	return &accessLogServer{
+		server: grpc.NewServer(),
+	}
+}
+
+func (a *accessLogServer) StreamAccessLogs(server v2.AccessLogService_StreamAccessLogsServer) error {
+	msg, err := server.Recv()
+	if err != nil {
+		return err
+	}
+
+	for _, httpLogEntry := range msg.LogEntries.(*v2.StreamAccessLogsMessage_HttpLogs).HttpLogs.LogEntry {
+		split := strings.Split(msg.GetIdentifier().GetLogName(), ";")
+		address := split[0]
+		format := split[1]
+		entry := formatEntry(httpLogEntry, format)
+		if err := a.sendLog(address, entry); err != nil {
+			return errors.Wrap(err, "could not send log")
+		}
+	}
+	return nil
+}
+
+func formatEntry(entry *envoy_data_accesslog_v2.HTTPAccessLogEntry, format string) string {
+	addrToString := func(addr *envoy_core.Address) string {
+		return fmt.Sprintf("%s:%d", addr.GetSocketAddress().GetAddress(), addr.GetSocketAddress().GetPortValue())
+	}
+	placeholders := map[string]string{
+		"%START_TIME%":                entry.GetCommonProperties().GetStartTime().Format(time.RFC3339),
+		"%DOWNSTREAM_REMOTE_ADDRESS%": addrToString(entry.GetCommonProperties().GetDownstreamRemoteAddress()),
+		"%DOWNSTREAM_LOCAL_ADDRESS%":  addrToString(entry.GetCommonProperties().GetDownstreamLocalAddress()),
+		"%UPSTREAM_LOCAL_ADDRESS%":    addrToString(entry.GetCommonProperties().GetUpstreamLocalAddress()),
+		"%UPSTREAM_CLUSTER%":          entry.GetCommonProperties().GetUpstreamCluster(),
```

Comment:
Although it might work fine, it looks inconsistent that `UPSTREAM_CLUSTER` is equal to Envoy `cluster name`, while `DOWNSTREAM_CLUSTER` is always a value of `service` tag

Reply:
Introduced additional variables expanded by Control Plane:
* `%KUMA_SOURCE_ADDRESS%`
* `%KUMA_SOURCE_SERVICE%`
* `%KUMA_DESTINATION_SERVICE%`

Introduced additional variables expanded by Access Logs Server:
* `%UPSTREAM_HOST%`
* `%UPSTREAM_REMOTE_ADDRESS%`
* `%DURATION%`
* `%RESPONSE_TX_DURATION%`

Reply:
fixed